### PR TITLE
Replace Geometry_cff with GeometryDB_cff in RecoEgamma

### PR DESCRIPTION
**Pull request by jeongeun**
**PR description:**
Review on the Reco part of https://github.com/cms-sw/cmssw/issues/31113

`process.load("Configuration.StandardSequences.Geometry_cff")`
 was outdated https://github.com/cms-sw/cmssw/pull/8810 
It should be replaced with
`process.load("Configuration.StandardSequences.GeometryDB_cff")`

In this PR, RecoEgamma configuration files (10 files) are fixed.
```
modified:   RecoEgamma/EgammaElectronProducers/test/egammaClustersToGsfElectrons_cfg.py
modified:   RecoEgamma/EgammaElectronProducers/test/egammaCoresToGsfElectrons_cfg.py
modified:   RecoEgamma/EgammaElectronProducers/test/egammaRawDataToGsfElectrons_cfg.py
modified:   RecoEgamma/EgammaElectronProducers/test/egammaRecHitsToElectronSeeds_cfg.py
modified:   RecoEgamma/EgammaElectronProducers/test/egammaRecHitsToGsfElectrons_cfg.py
modified:   RecoEgamma/EgammaIsolationAlgos/test/egammaIsoTest_cfg.py
modified:   RecoEgamma/EgammaIsolationAlgos/test/pfclusterIsoTest_cfg.py
modified:   RecoEgamma/EgammaIsolationAlgos/test/runInterestingEgammaIsoDetIdsSequence_cfg.py
modified:   RecoEgamma/ElectronIdentification/test/eidSelectorTest_cfg.py
modified:   RecoEgamma/ElectronIdentification/test/runElectronID_cfg.py
```

From 12_4_X to 12_5_X, following 4 files in Examples/test/ are all removed.
RecoEgamma/Examples/test/ElectronSeedMCAnalyzer_cfg.py
RecoEgamma/Examples/test/GsfElectronRedoFromCore_cfg.py
RecoEgamma/Examples/test/GsfElectronRedoFromRaw_cfg.py
RecoEgamma/Examples/test/SimplePi0DiscAnalyzer_cfg.py

### CODE REVIEW

File: RecoEgamma/ElectronIdentification/test/eidSelectorTest_cfg.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 process.load('FWCore/MessageService/MessageLogger_cfi')
```
Comment (clacaputo) [inline RecoEgamma/ElectronIdentification/test/eidSelectorTest_cfg.py:5]: Could you please also change
```suggestion
process.load('FWCore.MessageService.MessageLogger_cfi')
```

File: RecoEgamma/ElectronIdentification/test/eidSelectorTest_cfg.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
...
 process.load('Configuration/StandardSequences/Services_cff')
```
Comment (clacaputo) [inline RecoEgamma/ElectronIdentification/test/eidSelectorTest_cfg.py:6]:
```suggestion
process.load('Configuration.StandardSequences.Services_cff')
```

File: RecoEgamma/ElectronIdentification/test/runElectronID_cfg.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 process.load('FWCore/MessageService/MessageLogger_cfi')
```
Comment (clacaputo) [inline RecoEgamma/ElectronIdentification/test/runElectronID_cfg.py:5]:
```suggestion
process.load('FWCore.MessageService.MessageLogger_cfi')
```

File: RecoEgamma/ElectronIdentification/test/runElectronID_cfg.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
...
 process.load('Configuration/StandardSequences/Services_cff')
```
Comment (clacaputo) [inline RecoEgamma/ElectronIdentification/test/runElectronID_cfg.py:6]:
```suggestion
process.load('Configuration.StandardSequences.Services_cff')
```

**Review by clacaputo — CHANGES_REQUESTED** [review]

**Comment by clacaputo** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
+1

**Comment by tvami** [comment]
Let me comment that when the `GeometryDB` is used (as the name says) the geometry is taken from the DB, meaning that a GT will have to be specified. I see some of the modified files dont have that, for example
https://github.com/jeongeun/cmssw/blob/99ed83e1c8d3ee0474b5d347ea519ea34e22a1b2/RecoEgamma/EgammaIsolationAlgos/test/egammaIsoTest_cfg.py
so that's not expected to work after this migration. Some other files have very old GTs, which might work or not.

Given that fact that no wf has been broken that means these files are actually not used anywhere... but when/if they will be used, they'll need to have a GT in them

**Pull request merged**